src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -111,6 +111,43 @@ internal static HashAlgorithm TryGetAlgorithm(AssemblyHashAlgorithm algorithmId)
             }
         }
 
+        internal static IDisposable TryGetAlgorithmImpl(AssemblyHashAlgorithm algorithmId)
```

Comment:
There is no more specific type we can use. I'll move this method to Incremental hash, since it's not really general purpose. Comment added.